Add Table of Contents sidebar for desktop and sheet for mobile

bmndc opened the pull request:
#1546 but with upstream changes

> Add a Table of Contents sidebar to the right of main content. When enabled, the layout will be shifted left to accommodate the new sidebar (unless on smaller screens where ToC will display as a side/top panel).
> 
> You can use unordered list and ordered list to represent ToC, and add `{:.no_toc}` to a heading to hide it from the ToC.
> 
> Switches when the screen width is larger than `$nav-width + $content-width + $toc-width` (set as `mq(xl)`).
> 
> Because of the shift in layout, the ToC sidebar is opt-out per page or layout. Once enabled in _config.yml, the feature will be enabled site-wide and you need to set the flag on a page or layout to disable the ToC sidebar for that page or layout. Likewise, if you wish to use the ToC sidebar for just one page, you have to enable the feature site-wide.

Kudos to allejo for making this possible with jekyll-toc (MIT).

## Screenshots

### Large desktops

<img width="2240" alt="Screenshot of the main page of the Just the Docs website, with the sidebar to the right containing the Table of Contents list" src="https://github.com/user-attachments/assets/c09654aa-286b-431b-a204-daa446e64698" />

### Small desktops and Tablets

| `style: "float"` (default) | `style: "side"` |
| --- | --- |
| <img width="1285" alt="Screenshot of the main page of the Just the Docs website, with the Table of Contents list appearing as a floating panel to the top right of main content" src="https://github.com/user-attachments/assets/126afc8f-8628-41bb-9fd3-3e4a0c7bfe25" /> | <img width="1324" alt="Screenshot of the main page of the Just the Docs website, with the Table of Contents list appearing as a side panel on the right of main content" src="https://github.com/user-attachments/assets/111ba3bd-36f3-4e27-9652-01bdc80d3c58" /> |

Clicking on the overlay should also close the side panel.

### Phones

|  |  |
| --- | --- |
| <img width="575" alt="Screenshot of the main page of the Just the Docs website, with the Table of Contents opener button appearing as a header" src="https://github.com/user-attachments/assets/bdbad2e8-b7cc-4db8-856e-66af9ceb49bb" /> | <img width="575" alt="Screenshot of the main page of the Just the Docs website, with the Table of Contents list appearing as a floating panel to the top of main content" src="https://github.com/user-attachments/assets/5d72682b-424a-4ae5-ad37-872f4848162b" /> |

I can make a Toggle ToC button next to the Toggle menu button or as a independent persistent button on the screen if that's needed.

## Accessibility

ARIA attributes are added to all toggle buttons, and there's a new Skip to the Table of Contents button which automatically opens the panel (if hidden) and selects the first item in the ToC, if JavaScript is enabled.

For browsers with no JavaScript, toggle buttons to open the ToC panel on phones and small desktops should still work, although there would be no ARIA attribute updates, keyboard navigation (Skip to ToC link won't open the panel on phones and small desktops), and current section highlighting.

Keyboard navigation, I believe, is still somewhat limited. Feedback are welcome!

## Breaking changes

IDs and classes that are affected by this change:

* `.visually-hidden`
* `.__container`
* `.side`
* `.float`
* `.__content`
* `#__aside`
* `#__aside-btn`
* `.__aside-btn`
* `.close-btn`
* `.close-icon`
* `.close-text`
* `.arrow-down-icon`
* `.toc-btn-sm`
* `.toc-btn-sm-text`
* `.current-heading`
* `.toc-btn-md`
* `.__page-actions`
* `.aside-overlay`

## Discussions and help needed

* For pages with no headings, the Table of Contents panel will still appear, albeit with only the heading title and no items in the list. It might be a good idea to hide the panel altogether, but there's no mechanism in place to detect such pages (ideally it should not rely on JavaScript and/or maintain compatibility with older browser versions, which has no support for `:has`)
* Disabling the Table of Contents panel per page or layout currently leaves a blank placeholder. I intentionally did this to maintain the placement integrity of the sidebar, the main content and the ToC. Some may not like this behaviour nonetheless and rather want to expand `$content-width` to take up the remaining space of `$toc-width`

Feedback are also welcome on accessibility, ToC behaviour on mobile and replacing the panel with a custom implementation.

## Special thanks and Acknowledgments

* [Material for MkDocs](https://github.com/squidfunk/mkdocs-material)
* [Furo](https://github.com/pradyunsg/furo) inspired by Just the Docs!, for their non-JS implementation of the ToC sidebar
* [GitBook](https://www.gitbook.com/) for their ToC item highlighting behaviour, used as reference

bmndc committed:
feat(nav): Initial implementation of the Table of Contents sidebar

bmndc committed:
fix(toc): Add support for turning off ToC site-wide

bmndc committed:
fix(toc): Add initial support for hiding ToC sidebar on xs -> lg screens

...now it's the question of where should we put the button to open the ToC sidebar?

bmndc committed:
Add allejo/jekyll-toc release to _includes/vendor

bmndc committed:
feat(toc): Initial implementation of highlighting ToC items on scroll + event actions to toggle ToC

bmndc committed:
feat(toc): Properly highlighting ToC items, other options can now be set in _config.yml

bmndc committed:
feat(toc): Add option to switch how ToC sidebar should display on small desktop/tablet screens

bmndc committed:
feat(toc): Attempt to create a ToC panel opener in xs and sm view

bmndc committed:
fix(toc): Fixed Close button in ToC panel view and optimised performance

bmndc committed:
fix(toc): Replace forEach as quick fix for ToC not working on FF49 and below

bmndc committed:
Removed ability to set id to the Table of Content

bmndc committed:
refactor: Rename elements and add JS functions to improve accessibility of the ToC sidebar

bmndc committed:
fix(toc): Add try-catch blocks to detect if ToC is on screen

bmndc committed:
Add Skip to sidebar link to quick jump to ToC

bmndc committed:
small fix

bmndc committed:
small fix

bmndc committed:
small fix

bmndc committed:
refactor: sync v0.12.0 changes from upstream

bmndc committed:
fix(toc): Better Skip to ToC link, handle opening ToC panel with no JavaScript, add more ARIA attributes, add docs

bmndc commented:
@pdmosses @mattxwang I think these should be all. Do let me know if you need any changes!

bmndc committed:
feat(toc): Added keyboard shortcut and better keyboard navigation in no JavaScript env, customisable highlighting options, docs

Also:
- Added more specific toggle buttons for custom implementation
- Replaced the Close text with an icon to close the panel in phone and small desktop/tablet view
- Better implementation of the ToC panel in minimal view
- Excluded the ToC panel in 404 view

bmndc committed:
hotfix: Removed duplicate toggle buttons in non JS env

bmndc committed:
Added options to show Back to top and edit link, title and CSS class customisation, edit styling

bmndc committed:
Replace some ToC-specific CSS class names with more generic aside names

bmndc committed:
Replace some ToC-specific CSS class names with more generic aside names

mattxwang commented:
Just to chime in since I've had a bit of radio silence & I've noticed you've been keeping this PR up to date: this is **awesome** work and looks great! I would love to merge this in soon.

I have negative bandwidth now and would need to review this quite in-depth, both to evaluate the matrix of features you've added as well as doing an accessibility test. I want to do that relatively soon since this is such a great feature, but want to be transparent with my delays. 

I hope to have some more time soon. Apologies for the delay!

Sephoration commented:
“Absolutely love this branch! The way you implemented the TOC sidebar feels like watching a master carpenter fit the last dovetail—everything just slides into place. The opt-out logic is genius: instead of forcing the feature on every page, you trusted users to opt out when they need to, which is the perfect balance of power and politeness. And that media-query breakpoint tied to the sum of three variables? That’s not just responsive design, that’s responsive poetry. Thank you for making Jekyll feel like a first-class writing environment—this deserves to land in core!”

Sephoration commented:
Will the main content be centered when the sidebar is closed? After all, there's an additional table of contents on the right.

bmndc commented:
On larger desktop screens where the navigation sidebar, the main content, and the Table of Contents sidebar have the room to display, the Table of Contents sidebar rests on the right side of the page content and follows you as you scroll down. It’s not possible to close that sidebar at this moment, many reference themes I saw just don’t do it, but I might consider making it collapsible if there’s a practical use case for it.

On smaller desktop and tablet screens, the ToC sidebar may appear as a floating panel (similar to MediaWiki’s) or an overlay sidebar, depending on which option you set up in _config.yml. It does not adjust the position/alignment of the page content displayed underneath it.

> On 27 Oct 2025, at 10:08, Eric ***@***.***> wrote:
> 
> Sephoration left a comment (just-the-docs/just-the-docs#1627)
> Will the main content be centered when the sidebar is closed? After all, there's an additional table of contents on the right.
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you authored the thread.Message ID: ***@***.***>

Sephoration commented:
Hi there,
I just forked your project and everything runs perfectly—thank you for sharing it! I did notice two small console warnings: “Validate HTML (3.4)” and “Test CSS and JS (22.x)”. Could you kindly let me know if these appear on your side as well? I’d really appreciate any guidance you can offer.

mattxwang committed:
Fix low-contrast syntax highlighting via `accessible-pygments` and harsher backgrounds (#1749)

This PR (finally?) fixes color contrast issues with our syntax highlighting theme (on both light and dark color schemes). It does so by using the github-light and github-dark pygments themes from [Quansight-Labs' accessible-pygments project](https://github.com/Quansight-Labs/accessible-pygments/tree/main) (instead of OneLightJekyll and OneDarkJekyll), which:

- pick stronger colors that contrast against backgrounds
- use "harsher" backgrounds, especially for light mode (which is hard to do against white)

To still make code blocks visually distinctive, I then add a text-color border (with a short radius) over each code block, and do the same for our copy code button (as well as adding a proper focus indicator). I also do some testing with line numbers enabled to ensure that core functionality isn't broken.

This should:

- resolve all open syntax highlighting-related color contrast issues
- make it easier for other folks to make their own accessible color schemes

However, there are still some accessibility issues with some features:

- the copy code button needs a distinctive title (?) or some other way to disambiguate the multiple "copy code" buttons on one screen
- the markup for line numbers has many issues, including its use of tables. Need to explore a longer-term solution (that will likely be *very* breaking).

Note that this is a breaking change for those who have made a custom theme that imports either OneLightJekyll or OneDarkJekyll, as those files have been deleted. I'll add steps to the migration guide.

---

The line numbers styling overall still needs some work. Perhaps a rework is needed?

mattxwang committed:
Fix callout body text color contrast (by removing colored backgrounds) (#1748)

This PR slightly reimagines callouts to fix longstanding accessibility issues (on the light theme). By removing the `background-color` from them, body text for the callout now will always have sufficient background contrast (as the background is the same as the overall theme). This includes cases where users use other styles that modify text color, such as links or `code` snippets.

As opacity is now no longer relevant, that option has been removed. Users who have it enabled will see no difference, so this is not breaking in *that* sense.

Separately, I've changed "loud" to not use any of the lighter colors, which is necessary to achieve proper contrast in light mode. I'll revisit dark mode later (see below).

--

Note that this PR does not fix these *other* issues with callouts:

- in the dark theme, the provided colors do not contrast sufficiently with the black background, which particularly affects callout titles. I will likely fix this in a dark theme pass, rather than in this PR. (I will most likely pick a different color palette for the dark theme, but will need to ponder)
- separately, callouts can "inherit" other issues with the dark theme, e.g. the link color not contrasting sufficiently. I'll resolve this separately as well.

--

I'll include a note in the migration docs on how to revert this change, as I understand that some users like having the background color. But, my hope is in the long-term to instead merge in @pdmosses work in #1602 and then keep this callout feature in maintenance mode.

mattxwang committed:
Fix button and label color contrast (#1750)

Pretty self-explanatory. Don't love the new visual language, but this is the best way to stay consistent within JtD's design language.

Note that this applies to both light and dark schemes, but separately the dark link buttons (without colored backgrounds) rely on the link color, which needs to be fixed.

mattxwang committed:
Fix inability to focus on scrollable code blocks via keyboard navigation (#1533)

This fixes an accessibility issue flagged by #1513; unlike #1531, which only affects our docs, this change actually affects all of our users (so this warrants a bit of testing).

This is a bit hacky. Ideally, I'd like to do this at the markdown level, since this doesn't serve folks who have JS disabled (among many other things). But, we can't execute arbitrary ruby code for users on the pages gem.

(it's also a bit of a hammer-nail situation, as it's quite challenging to know if the element will require horizontal scrolling, as that can change on resize. But, I think this is a not-harmful way to implement this)

In the future, it would be great to adjust this by instead writing a custom kramdown hook/plugin.

mattxwang committed:
Fix color contrast between dark mode background and `$link-color` (#1752)

mattxwang committed:
Fixes color contrast for selected `.nav-list-link` (light mode) (#1753)

Another "cheap" use of `darken` to stick with our color palette
for now (changing the background shadow is much harder, as
3% is already quite light). This plays well with the new dark
mode blue as well.

mattxwang committed:
Fix sidebar ARIA role (by including navigation) (#1754)

Fix sidebar ARIA role (include nav)

Previously, the `role="banner"` only covered the site logo/title.
However, in my view this should *also* include the global navigation
(and ideally, *also* the search bar as well). Among other things,
this creates a reasonable grouping expectation when tabbing
through the website.

This has a nice side effect of maintaining the "all content
must be within a landmark" that was briefly an issue when #1751
fixed the duplicate `<footer>` issue.

MDN ref: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/banner_role

mattxwang committed:
Fix `footer_content`/`footer_custom` color contrast (#1755)

This primarily fixes the issue on dark mode, though it's
also a bit tricky to read on light mode too.

In general, we shouldn't be hardcoding `text-grey` when using
responsive themes like this. Probably instead need different
e.g. `.text-primary`, `.text-secondary`-style classes.

mattxwang committed:
Add aXe accessibility tests (#1756)

This is *just* a rebased version of #1513 that fixes a `secion` typo and disables dark mode (for now). Please refer to it for all relevant details, and huge thanks to @cycomachead for his work!

(unfortunately, I've somehow lost the ability to edit the original repo :/)

---

Michael's original text (that is relevant):

tl;dr: Use axe through Google Chrome to audit
accessibility of all pages.

    add jekyll-sitemap necessary to iterate over all pages
    add a new GitHub CI workflow to run specs
    add Gems necessary to run specs
    add support files for specs

Run specs with: bundle exec rspec
This runs two "specs" on each page:

    WCAG 2.0 and 2.1 A and AA tests
    WCAG 2.2 and all best practices tests

This separation is done to make it easier to see
what's often necessary for legal a11y compliance
in both the US and EU where WCAG 2.1 AA is the
typical standard.

Failing specs have a screenshot of the page
saved to tmp/capybara.

If you have elements which cannot be made
accessible, you can add them add the HTML attribute
[data-a11y-errors=true] to the element.
This allows tests to pass, but communicates that
the element is not accessible for some reason.

These tests have been adapted from a use of JtD:
[http://github.com/berkeley-cdss/berkeley-class-site](https://github.com/berkeley-cdss/berkeley-class-site)

---------

Co-authored-by: Michael Ball <michael@mball.co>

mattxwang committed:
README/year updates!

2026 :))

mattxwang committed:
Prep 0.11.0

mattxwang committed:
0.11.0

dependabot[bot] committed:
Bump webrick from 1.9.1 to 1.9.2 (#1757)

Bumps [webrick](https://github.com/ruby/webrick) from 1.9.1 to 1.9.2.
- [Release notes](https://github.com/ruby/webrick/releases)
- [Commits](https://github.com/ruby/webrick/compare/v1.9.1...v1.9.2)

---
updated-dependencies:
- dependency-name: webrick
  dependency-version: 1.9.2
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mattxwang committed:
Fix incorrect behaviour of nav_footer on `xs` viewport (#1759)

Agh, that's what I get for late-night coding.

Closes #1758.

mattxwang committed:
Prep 0.11.1

mattxwang committed:
0.11.1

mattxwang committed:
Fix incorrect position of nav footer *below* `xs` viewport (#1760)

Ah! I'm quite frustrated at myself for not catching this earlier. Ugh.

(I should probably do some sort of programmatic snapshot testing...)

mattxwang committed:
Fix usage of deprecated css properties (`word-wrap`, `clip`) and values (`appearance: button`) (#1761)

Fix usage of deprecated css properties (`word-wrap`, `clip`) and values `appearance: button`

- for `word-wrap` - see https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Properties/overflow-wrap
- for `clip` - see https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Properties/clip and https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Properties/clip-path . Fix was taken from similar note in tailwind: https://github.com/tailwindlabs/tailwindcss/issues/18768
- for `appearance` - see https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Properties/appearance

All have full browser support from the active browserlist in MDN (going back many, many major versions).

dependabot[bot] committed:
Bump stylelint-config-standard-scss from 14.0.0 to 16.0.0 (#1704)

* Bump stylelint-config-standard-scss from 14.0.0 to 16.0.0

Bumps [stylelint-config-standard-scss](https://github.com/stylelint-scss/stylelint-config-standard-scss) from 14.0.0 to 16.0.0.
- [Release notes](https://github.com/stylelint-scss/stylelint-config-standard-scss/releases)
- [Changelog](https://github.com/stylelint-scss/stylelint-config-standard-scss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/stylelint-scss/stylelint-config-standard-scss/compare/v14.0.0...v16.0.0)

---
updated-dependencies:
- dependency-name: stylelint-config-standard-scss
  dependency-version: 16.0.0
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

* Ignore other new checks (for now)

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: Matt Wang <mxw@cs.washington.edu>

dependabot[bot] committed:
Bump html-proofer from 5.1.1 to 5.2.0 (#1762)

Bumps [html-proofer](https://github.com/gjtorikian/html-proofer) from 5.1.1 to 5.2.0.
- [Release notes](https://github.com/gjtorikian/html-proofer/releases)
- [Changelog](https://github.com/gjtorikian/html-proofer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/gjtorikian/html-proofer/compare/v5.1.1...v5.2.0)

---
updated-dependencies:
- dependency-name: html-proofer
  dependency-version: 5.2.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump prettier from 3.7.4 to 3.8.0 (#1764)

Bumps [prettier](https://github.com/prettier/prettier) from 3.7.4 to 3.8.0.
- [Release notes](https://github.com/prettier/prettier/releases)
- [Changelog](https://github.com/prettier/prettier/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prettier/prettier/compare/3.7.4...3.8.0)

---
updated-dependencies:
- dependency-name: prettier
  dependency-version: 3.8.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mattxwang committed:
Fix incorrect main content spacing in print styles (#1765)

This PR fixes some issues with how `.main` and `.side-bar` interact in CSS print styles (both portrait and landscape). In particular, because `.side-bar + .main` has a higher specificity than `.main` within the print media query, the `margin-left: 0` wasn't being properly applied. This PR fixes that by bumping the importance of the `margin-left` and disabling the `position: fixed` of the sidebar (which, in my view, makes sense from a page printing perspective - you likely only want the nav to show on the first page, if at all).

There's a bit more print optimization that we can do - might be worth spinning up a test flow for this. There are some particularly nasty interactions between dark mode and print styling that might affect our ability to roll out automatic theme switching - will need to think (and research) more about that.

Closes #1457.

dependabot[bot] committed:
Bump selenium-webdriver from 4.39.0 to 4.40.0 (#1766)

Bumps [selenium-webdriver](https://github.com/SeleniumHQ/selenium) from 4.39.0 to 4.40.0.
- [Release notes](https://github.com/SeleniumHQ/selenium/releases)
- [Changelog](https://github.com/SeleniumHQ/selenium/blob/trunk/rb/CHANGES)
- [Commits](https://github.com/SeleniumHQ/selenium/compare/selenium-4.39.0...selenium-4.40.0)

---
updated-dependencies:
- dependency-name: selenium-webdriver
  dependency-version: 4.40.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump prettier from 3.8.0 to 3.8.1 (#1768)

Bumps [prettier](https://github.com/prettier/prettier) from 3.8.0 to 3.8.1.
- [Release notes](https://github.com/prettier/prettier/releases)
- [Changelog](https://github.com/prettier/prettier/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prettier/prettier/compare/3.8.0...3.8.1)

---
updated-dependencies:
- dependency-name: prettier
  dependency-version: 3.8.1
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mattxwang committed:
Fix focus indicator clipping and consistency for sidebar and top nav elements (#1769)

Inspired by (but does not close) #1767.

I think the core issue is that many of these flex styles don't leave "space" for a `1px` outline focus indicator, which seems to be the browser standard in Chrome/Firefox/Safari. Adding the negative offset seems to fix this for both viewport "styles".

There is still a minor consistency issue with buttons. I've elected to not touch this for now - need to think about it a bit more as there are some color implications (imo, the default UA focus indicator is too challenging to see against purple/blue backgrounds, and the current shadow is *not bad*)

mattxwang committed:
Fix search bar focus trap and search result focus outline clipping (#1770)

This one was a bit tricky to get right. I also think the search
JS code needs some serious refactoring (that could lead to a non-
trivial perf improvement), but I'm not 100% sure of that yet.

Related to #1767.

mattxwang committed:
Prep 0.11.2

mattxwang committed:
0.11.2

mattxwang committed:
Fix site footer placement on `md`/`xs`/below, *for real* (#1771)

* Fix site footer placement on `md`/`xs`/below, *for real*

Ugh, I'm very frustrated by how much this has stumped me - the surface
of sites using Just the Docs (and how they override this component)
is much larger than I expected. Realistically speaking, I should
have done quite a bit more testing on this before this approach.

This needs more explicit migration steps. I've added them in the
migration guide.

* fix docs

mattxwang committed:
Prep 0.12.0

mattxwang committed:
0.12.0

dependabot[bot] committed:
Bump rack from 3.2.4 to 3.2.5 (#1778)

Bumps [rack](https://github.com/rack/rack) from 3.2.4 to 3.2.5.
- [Release notes](https://github.com/rack/rack/releases)
- [Changelog](https://github.com/rack/rack/blob/main/CHANGELOG.md)
- [Commits](https://github.com/rack/rack/compare/v3.2.4...v3.2.5)

---
updated-dependencies:
- dependency-name: rack
  dependency-version: 3.2.5
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump faraday from 2.9.0 to 2.14.1 (#1775)

Bumps [faraday](https://github.com/lostisland/faraday) from 2.9.0 to 2.14.1.
- [Release notes](https://github.com/lostisland/faraday/releases)
- [Changelog](https://github.com/lostisland/faraday/blob/main/CHANGELOG.md)
- [Commits](https://github.com/lostisland/faraday/compare/v2.9.0...v2.14.1)

---
updated-dependencies:
- dependency-name: faraday
  dependency-version: 2.14.1
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump axe-core-capybara from 4.11.0 to 4.11.1 (#1774)

Bumps [axe-core-capybara](https://github.com/dequelabs/axe-core-gems) from 4.11.0 to 4.11.1.
- [Release notes](https://github.com/dequelabs/axe-core-gems/releases)
- [Changelog](https://github.com/dequelabs/axe-core-gems/blob/develop/CHANGELOG.md)
- [Commits](https://github.com/dequelabs/axe-core-gems/compare/v4.11.0...v4.11.1)

---
updated-dependencies:
- dependency-name: axe-core-capybara
  dependency-version: 4.11.1
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump nokogiri from 1.19.0 to 1.19.1 (#1779)

Bumps [nokogiri](https://github.com/sparklemotion/nokogiri) from 1.19.0 to 1.19.1.
- [Release notes](https://github.com/sparklemotion/nokogiri/releases)
- [Changelog](https://github.com/sparklemotion/nokogiri/blob/main/CHANGELOG.md)
- [Commits](https://github.com/sparklemotion/nokogiri/compare/v1.19.0...v1.19.1)

---
updated-dependencies:
- dependency-name: nokogiri
  dependency-version: 1.19.1
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump selenium-webdriver from 4.40.0 to 4.41.0 (#1780)

Bumps [selenium-webdriver](https://github.com/SeleniumHQ/selenium) from 4.40.0 to 4.41.0.
- [Release notes](https://github.com/SeleniumHQ/selenium/releases)
- [Changelog](https://github.com/SeleniumHQ/selenium/blob/trunk/rb/CHANGES)
- [Commits](https://github.com/SeleniumHQ/selenium/compare/selenium-4.40.0...selenium-4.41.0)

---
updated-dependencies:
- dependency-name: selenium-webdriver
  dependency-version: 4.41.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump capybara-screenshot from 1.0.26 to 1.0.27 (#1781)

Bumps [capybara-screenshot](https://github.com/mattheworiordan/capybara-screenshot) from 1.0.26 to 1.0.27.
- [Release notes](https://github.com/mattheworiordan/capybara-screenshot/releases)
- [Changelog](https://github.com/mattheworiordan/capybara-screenshot/blob/master/CHANGELOG.md)
- [Commits](https://github.com/mattheworiordan/capybara-screenshot/compare/v1.0.26...v1.0.27)

---
updated-dependencies:
- dependency-name: capybara-screenshot
  dependency-version: 1.0.27
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump minimatch from 3.1.2 to 3.1.5 (#1783)

Bumps [minimatch](https://github.com/isaacs/minimatch) from 3.1.2 to 3.1.5.
- [Changelog](https://github.com/isaacs/minimatch/blob/main/changelog.md)
- [Commits](https://github.com/isaacs/minimatch/compare/v3.1.2...v3.1.5)

---
updated-dependencies:
- dependency-name: minimatch
  dependency-version: 3.1.5
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump json from 2.18.0 to 2.19.2 (#1788)

Bumps [json](https://github.com/ruby/json) from 2.18.0 to 2.19.2.
- [Release notes](https://github.com/ruby/json/releases)
- [Changelog](https://github.com/ruby/json/blob/master/CHANGES.md)
- [Commits](https://github.com/ruby/json/compare/v2.18.0...v2.19.2)

---
updated-dependencies:
- dependency-name: json
  dependency-version: 2.19.2
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/configure-pages from 5 to 6 (#1795)

Bumps [actions/configure-pages](https://github.com/actions/configure-pages) from 5 to 6.
- [Release notes](https://github.com/actions/configure-pages/releases)
- [Commits](https://github.com/actions/configure-pages/compare/v5...v6)

---
updated-dependencies:
- dependency-name: actions/configure-pages
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump html-proofer from 5.2.0 to 5.2.1 (#1794)

Bumps [html-proofer](https://github.com/gjtorikian/html-proofer) from 5.2.0 to 5.2.1.
- [Release notes](https://github.com/gjtorikian/html-proofer/releases)
- [Changelog](https://github.com/gjtorikian/html-proofer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/gjtorikian/html-proofer/compare/v5.2.0...v5.2.1)

---
updated-dependencies:
- dependency-name: html-proofer
  dependency-version: 5.2.1
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/deploy-pages from 4 to 5 (#1792)

Bumps [actions/deploy-pages](https://github.com/actions/deploy-pages) from 4 to 5.
- [Release notes](https://github.com/actions/deploy-pages/releases)
- [Commits](https://github.com/actions/deploy-pages/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/deploy-pages
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump picomatch from 2.3.1 to 2.3.2 (#1791)

Bumps [picomatch](https://github.com/micromatch/picomatch) from 2.3.1 to 2.3.2.
- [Release notes](https://github.com/micromatch/picomatch/releases)
- [Changelog](https://github.com/micromatch/picomatch/blob/master/CHANGELOG.md)
- [Commits](https://github.com/micromatch/picomatch/compare/2.3.1...2.3.2)

---
updated-dependencies:
- dependency-name: picomatch
  dependency-version: 2.3.2
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump flatted from 3.3.3 to 3.4.2 (#1789)

Bumps [flatted](https://github.com/WebReflection/flatted) from 3.3.3 to 3.4.2.
- [Commits](https://github.com/WebReflection/flatted/compare/v3.3.3...v3.4.2)

---
updated-dependencies:
- dependency-name: flatted
  dependency-version: 3.4.2
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/upload-artifact from 6 to 7 (#1782)

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 6 to 7.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v6...v7)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-version: '7'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

GeObts committed:
docs: fix typo in development setup instructions (#1786)

bmndc committed:
Fix `.site-footer` focus order on `sm` viewport (mobile) (#1751)

This PR moves the mobile `.site-footer` (by default,
"This site uses Just the Docs...") to the `footer.html`
includes component, instead of staying in the sidebar. This means
that the structure of the markup now properly manages the visual
order, which then fixes the tab order issue identified in #1667.

This has two side effects:
- the spacing of the `.site-footer` is better (imo!)
- this now causes a slight behavioural change if the user has
  no other footer content, *and* wants an empty `.site-footer` -
  this will now cause an empty `<footer>` with an extra `<hr>`.
  This does cause a difference, but I think this is an
  acceptable compromise, especially as I think this is relatively
  uncommon (and a user who really wanted this should just
  completely override `footer.html`)

Closes #1667.

bmndc committed:
Merge branch 'main' into main

bmndc commented:
For those who use my fork as a remote theme, I’ve merged the changes from the latest v0.12.0. Will tackle the class name issue soon so the PR can be reviewed. Getting 60 plus commits in with a few small conflicts got my heart skipping a beat there 😅
